docs/pip.md

```diff
@@ -0,0 +1,163 @@
+# Pip
+
+This document describes some of the more intricate details of Cachito support for pip. For a high
+level overview, look [here in the README](../README.md#pip).
+
+Cachito has a number of specific requirements when it comes pip packages. Some of those stem from
+the general ideas behind Cachito (e.g. reproducibility), some from the technical challenges of
+supporting a packaging system which defines most metadata through a Python executable. Read on for
+more details.
+
+## `requirements.txt`
+
+One of the main component of a pip package is the `requirements.txt` file. Typically, the file might
+look something like this:
+
+```ini
+requests
+git+https://github.com/containerbuildsystem/dockerfile-parse
+https://github.com/containerbuildsystem/operator-manifest/archive/v0.0.3.zip
+```
+
+The dependencies in this file are:
+
+* `requests`, a PyPI dependency
+* `dockerfile-parse`, a Git dependency
+* `operator-manifest`, an HTTPS dependency
+
+Git and HTTP(S) dependencies will henceforth collectively be referred to as "external."
+
+### Pinning versions
+
+To make sure builds are reproducible, Cachito will require that all dependencies be pinned to
+a specific version.
+
+* for PyPI dependencies, use the `==` operator: `requests==2.24.0`
+* for Git dependencies, specify the commit hash:
+  * `git+https://github.com/containerbuildsystem/dockerfile-parse@ed38dc2eadf6890b524340f45a7d172d02196e61`
+* for HTTP(S) dependencies, include the hash of the source archive (truncated for readability):
+  * `https://github.com/containerbuildsystem/operator-manifest/archive/v0.0.3.zip#cachito_hash=sha256:7774db0`
+
+### Recursive dependencies
+
+You will need to include not only the direct dependencies for your application, but all the
+dependencies of those dependencies (recursively) as well. There are two reasons for this. The first
+one is reproducibility (you will need to pin the versions of recursive dependencies as well). The
+second reason is that, to determine the dependencies of your dependencies, Cachito would have to
+execute their `setup.py` files, which is not desirable.
+
+### `pip-compile`
```

Comment:
(optional): I do understand the reason behind having quotes here to make them codable but when you look at the README, it just looks bit off appearance wise. I'd rather just have it as regular title or put them in normal non-codable quotes.